Nicsaal/continuous gps

nicsaal opened the pull request:
Please review.

oliversaal commented on BasicLocationSample/app/gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -0,0 +1,6 @@
+#Mon Jun 12 15:25:49 CAT 2017
```
I think all the files that placed under BasicLocationSample/app/* should not be checked/committed. These are files that are generated by the build system for the app but they shouldn't be checked in IMO. Can you "revert" these BasicLocationSample/app/* files from this PR. Give me a shout if you need any help with it. It's a good skill to know.

oliversaal replied:
Actually, I need to correct something - the files that need to be reverted are gradle-related. So revert only the BasicLocationSample/app/gradle* additions.

oliversaal commented on BasicLocationSample/app/src/main/AndroidManifest.xml:
```diff
@@ -26,6 +26,7 @@ limitations under the License.
     an accuracy approximately equivalent to a city block.
     -->
     <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/>
```
If we're now opting only for the fine-grain location, we should probably remove the permission for the COARSE_LOCATION

nicsaal replied:
done

oliversaal commented on BasicLocationSample/app/src/main/java/com/google/android/gms/location/sample/basiclocationsample/MainActivity.java:
```diff
@@ -39,49 +51,187 @@
+                settingStates = result.getLocationSettingsStates();
+                switch (status.getStatusCode()) {
+                    case LocationSettingsStatusCodes.SUCCESS:
+                        // Info message -log file
```
Nit - "Info message - log file". Can you search and replace all occurrences in this file with "- log" instead of "-log"

nicsaal replied:
done

oliversaal commented on BasicLocationSample/app/src/main/java/com/google/android/gms/location/sample/basiclocationsample/MainActivity.java:
```diff
@@ -16,9 +16,9 @@
+import android.content.IntentSender;
 import android.location.Location;
 import android.os.Bundle;
-import android.support.v7.app.ActionBarActivity;
```
Was the methods provided by ActionBarActivity ever used - just asking since I didn't see any reference to it. Could it be that this was provided by the original sample with an action bar menu with menu options to start/stop the GPS updates?

nicsaal replied:
Correct the original sample makes use of an actionbar. I removed these imports because we're not using it.

oliversaal commented on BasicLocationSample/app/src/main/java/com/google/android/gms/location/sample/basiclocationsample/MainActivity.java:
```diff
@@ -39,49 +51,187 @@
+    /**
+     * Location settings result variables
+     */
+    protected Status status;
```
Since these are member variables of the class MainActivity, we should follow the established convention and call them "mStatus" and "mSettingsStates" below. You could even issue a PR to Google's original sample to fix this. :-)

oliversaal replied:
Don't forget to "Squash and merge" your changes with an appropriate description of the feature change (e.g. Not a generic 'New Feature Added')

nicsaal replied:
done

oliversaal commented on BasicLocationSample/app/src/main/java/com/google/android/gms/location/sample/basiclocationsample/MainActivity.java:
```diff
@@ -39,49 +51,187 @@
+                        try {
+                            status.startResolutionForResult(MainActivity.this, REQUEST_CHECK_SETTINGS);
+                        }
+                        catch(IntentSender.SendIntentException e){
```
Move the catch up a line so it reads like this: } catch (Intent ..... e) {

nicsaal replied:
done

nicsaal commented:
Improved code to continuously update GPS location and added a timestamp of when the last GPS coordinates was retrieved. Also removed COARSE permissions from manifest.xml file.

nicsaal commented:
Improved code to continuously update GPS location and added a timestamp of when the last GPS coordinates was retrieved. Also removed COARSE permissions from manifest.xml file.
